src/registry/domain/options-sanitiser.js

```diff
@@ -53,11 +53,38 @@ module.exports = function(input) {
   }
 
   options.customHeadersToSkipOnWeakVersion = (options.customHeadersToSkipOnWeakVersion ||
-    [])
-    .map(s => s.toLowerCase());
+    []
+  ).map(s => s.toLowerCase());
 
   options.port = process.env.PORT || options.port;
   options.timeout = options.timeout || 1000 * 60 * 2;
 
+  if (options.s3) {
+    options.storage = {};
+    options.storage.adapterType = 's3';
+    options.storage.options = options.s3;
+  }
+
+  if (options.refreshInterval && options.storage) {
+    options.storage.options.refreshInterval = options.refreshInterval;
+  }
+
+  if (options.verbosity && options.storage) {
+    options.storage.options.verbosity = options.verbosity;
+  }
+  if (
+    options.storage &&
+    options.storage.adapterType &&
+    !options.storage.adapter
+  ) {
+    switch (options.storage.adapterType) {
+    case 's3':
```

Comment:
As mentioned above this will avoid to have a switch statement on the `adapterType`, and just setup a default in case for `s3` config (legacy) overridable with the new API.